src/tensor_product.jl

```diff
@@ -0,0 +1,19 @@
+# This files defines an interface for the tensor product of two axes
+# https://en.wikipedia.org/wiki/Tensor_product
+
+⊗() = tensor_product()
+⊗(a) = tensor_product(a)
+⊗(a1, a2) = tensor_product(a1, a2)  # default
+⊗(a1, a2, as...) = ⊗(⊗(a1, a2), as...)   # allow to specialize ⊗(a1, a2) to fusion_product
```

Comment:
Can you move the comments to be on their own lines above the function definitions? I prefer that style in general.